controllers/machine_controller.go

```diff
@@ -268,9 +266,19 @@ func (r *MachineReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cl
 			}
 			continue
 		}
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
 
+	resNode, err := r.reconcileNode(ctx, cluster, m)
+	if err != nil {
 		errs = append(errs, err)
 	}
+	if resNode.Requeue {
+		res = resNode
```

Comment:
I think we should wait for https://github.com/kubernetes-sigs/cluster-api/pull/3723 to get merged before this PR so we can have cleaner handling of return values.
Also this implementation most probably is missing util.LowestNonZeroResult here